docs/node/web3.md

```diff
@@ -0,0 +1,266 @@
+---
+description: Web3 gateway for Emerald and Sapphire ParaTimes
+---
+
+# Oasis Web3 Gateway for your EVM ParaTime
+
+This guide will walk you through the steps needed to set up the Oasis Web3
+gateway for EVM-compatible ParaTimes, such as Emerald and Sapphire.
+
+:::caution
+
+Each ParaTime requires its own instance of the Web3 gateway!
+
+:::
+
+## Prerequisites
+
+### Hardware
+
+In addition to the minimum hardware requirements for running the Oasis node,
+the following should be added for running the Web3 gateway:
+
+* CPU:
+  * Minimum: 2.0 GHz x86-64 CPU
+  * Recommended: 2.0 GHz x86-64 CPU with 2 cores/vCPUs
+* Memory:
+  * Minimum: 4 GB of ECC RAM
+  * Recommended: 8 GB of ECC RAM
+* Storage:
+  * Minimum: 300 GB of SSD or NVMe fast storage
+  * Recommended: 500 GB of SSD or NVMe fast storage
+
+:::info
+
+To put the figures above into perspective, the Web3 gateway for Emerald with
+PostgreSQL encountered **210 GB** of database growth in ~5 months between
+Nov 18, 2021 and Apr 11, 2022 (since the [Emerald Mainnet launch]).
+
+:::
+
+### Oasis ParaTime Client Node
+
+The Web3 gateway requires a locally deployed ParaTime-enabled Oasis Node.
+First, follow the
+[Oasis ParaTime client node](run-your-node/paratime-client-node.mdx)
+guide on how to configure the Oasis client node with one or more ParaTimes.
+Always use the exact combination of the Oasis node/ParaTime versions as
+published on the network parameters page for either [Mainnet] or [Testnet]. 
+
+Apart from the transactions that happen on-chain and produce some effects,
+there are also a number of read-only queries implemented in the Oasis protocol
+and EVM. Some of them may be quite resource-hungry such as simulating EVM calls
+and are disabled by default to avoid DDOS attacks. If your Oasis node instance
+will only be used by you and your Web3 gateway(s), you can safely enable
+*expensive* transactions by adding the following to your Oasis node config:
+
+```yaml title='config.yml'
+# ... sections not relevant are omitted ...
+runtime:
+  mode: client
+  paths:
+    - {{ emerald_bundle_path }}
+    - {{ sapphire_bundle_path }}
+
+  config:
+    "{{ emerald_paratime_id }}":
+      estimate_gas_by_simulating_contracts: true
+      allowed_queries:
+        - all_expensive: true
+    "{{ sapphire_paratime_id }}":
+      estimate_gas_by_simulating_contracts: true
+      allowed_queries:
+        - all_expensive: true
+```
+
+In the config above replace `{{ ... }}` placeholders with actual Paratime IDs:
+* `{{ emerald_paratime_id }}`:
+  * Emerald on Mainnet: `000000000000000000000000000000000000000000000000e2eaa99fc008f87f`
+  * Emerald on Testnet: `00000000000000000000000000000000000000000000000072c8215e60d5bca7`
+* `{{ sapphire_paratime_id }}`:
+  * Sapphire on Testnet: `000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c`
```

Comment:
```suggestion
  * Emerald on [Mainnet]: `000000000000000000000000000000000000000000000000e2eaa99fc008f87f`
  * Emerald on [Testnet]: `00000000000000000000000000000000000000000000000072c8215e60d5bca7`
* `{{ sapphire_paratime_id }}`:
  * Sapphire on [Testnet]: `000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c`
```
What do you think about linking to network parameters pages in this way?

Reply:
Yeah, good idea.